### PR TITLE
removed invert function

### PR DESCRIPTION
**Pull request by martinhruska**
Based on dicussion in issue #33 this PR removes `invert` function and keep `revert` function (with semantics of removed invert function).
Rationalization: `revert` is a standard name in automato libraries and was implemented by Ondra to data structures initally used in this library. The `invert` function was orignally created by Lukas and Juraj in their library for data structures which were later adopted in the Mata library. Therefore we take behaviour of `invert` but keep the name `revert` which is standard.

### CODE REVIEW

File: src/nfa/nfa.cc [hunk 1]
```diff
@@ -593,15 +593,17 @@ void Mata::Nfa::revert(Nfa* result, const Nfa& aut)
 
     if (aut.get_num_of_states() > result->get_num_of_states()) { result->increase_size(aut.get_num_of_states()); }
 
-    result->initialstates = aut.finalstates;
-    result->finalstates = aut.initialstates;
-
-    for (size_t i = 0; i < aut.get_num_of_states(); ++i)
+    for (State sourceState = 0; sourceState < aut.transitionrelation.size(); ++sourceState)
```
Comment (Adda0) [inline src/nfa/nfa.cc:596]: When we are at it, could we extract the call for `aut.transitionrelation.size()` into a constant? We already do it everywhere, so…
```suggestion
    const size_t num_of_states{ aut.transitionrelation.size() };
    for (State sourceState = 0; sourceState < num_of_states; ++sourceState)
```

**Review by Adda0 — COMMENTED** [review]
LGTM. However, are you sure there are no more calls to `invert()` left in the code (and in the tests)? This applies to the Python binding as well (if I remember correctly, there is a binding for invert, too). Should I change the Python binding, or do you want to update it yourself?

**Comment by martinhruska** [comment]
If you can change python binding, please do it.

**Comment by Adda0** [comment]
I will make a new PR onto this PR. Sounds good to you?

**Comment by tfiedor** [comment]
Let's not stack PRs, please. Since, there will not be any conflicts, I think it is safe to push directly to this branch.

**Review by tfiedor — APPROVED** [review]
Yeah, sure, dope PR, just fix it in the binding.

**Comment by Adda0** [comment]
That could be done, too. However, as it turns out, there is no work to be done here, as there is only `revert()` in the Python binding already.

**Review by Adda0 — APPROVED** [review]
Feel free to merge when you are ready.

**Comment by tfiedor** [comment]
I see, obviously, I expertly foresaw the future change. All hail the mighty sea lion!

**Comment by Adda0** [comment]
Exactly.

**Comment by martinhruska** [comment]
I like these situations when things take care about themself. One task already done today, no activity needed.

**Pull request merged**